snuba/processor.py

```diff
@@ -36,7 +36,7 @@ class MessageProcessor(ABC):
     """
 
     @abstractmethod
-    def process_message(self, message, metadata=None) -> Optional[ProcessedMessage]:
+    def process_message(self, message, metadata) -> Optional[ProcessedMessage]:
```

Comment:
Trying to annotate this `metadata` argument with the correct type causes a circular import, unfortunately. Since this parameter is going away in the next step of this process, I didn't take the time to try and work around that.